card/src/main/java/com/adyen/checkout/card/CardView.kt

```diff
@@ -0,0 +1,318 @@
+ *
+ * Created by ran on 13/3/2019.
+ */
+package com.adyen.checkout.card
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.res.TypedArray
+import android.text.Editable
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.OnFocusChangeListener
+import android.view.WindowManager
+import androidx.annotation.StringRes
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.Observer
+import com.adyen.checkout.card.data.CardType
+import com.adyen.checkout.card.data.DetectedCardType
+import com.adyen.checkout.card.data.ExpiryDate
+import com.adyen.checkout.card.databinding.CardViewBinding
+import com.adyen.checkout.card.ui.SecurityCodeInput
+import com.adyen.checkout.components.api.ImageLoader
+import com.adyen.checkout.components.ui.view.AdyenLinearLayout
+import com.adyen.checkout.components.ui.view.AdyenTextInputEditText
+import com.adyen.checkout.components.ui.view.RoundCornerImageView
+import com.adyen.checkout.components.validation.ValidatedField
+
+/**
+ * CardView for [CardComponent].
+ */
+@Suppress("TooManyFunctions")
+class CardView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
+    AdyenLinearLayout<CardOutputData?, CardConfiguration?, CardComponentState?, CardComponent?>(context, attrs, defStyleAttr),
+    Observer<CardOutputData?> {
+
+    private val binding: CardViewBinding = CardViewBinding.inflate(LayoutInflater.from(context), this)
+
+    private val mCardInputData = CardInputData()
+    private var mImageLoader: ImageLoader? = null
+
+    init {
+        orientation = VERTICAL
+        val padding = resources.getDimension(R.dimen.standard_margin).toInt()
+        setPadding(padding, padding, padding, 0)
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        // Prevent taking screenshot and screen on recents.
+        getActivity(context)?.window?.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        getActivity(context)?.window?.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
+    }
+
+    override fun initView() {
+        initCardNumberInput()
+        initExpiryDateInput()
+        initSecurityCodeInput()
+        initHolderNameInput()
+
+        binding.switchStorePaymentMethod.setOnCheckedChangeListener { _, isChecked ->
+            mCardInputData.isStorePaymentSelected = isChecked
+            notifyInputDataChanged()
+        }
+        if (component.isStoredPaymentMethod()) {
+            component.getStoredPaymentInputData()?.let {
+                setStoredCardInterface(it)
+            }
+        } else {
+            binding.textInputLayoutCardHolder.visibility = if (component.isHolderNameRequire()) VISIBLE else GONE
```

Comment:
What do you think if we add an extension function like this:

```kotlin
fun View.setVisible(visible: Boolean) {
    visibility = if (visible) View.VISIBLE else View.GONE
}
```

This way we can just use `binding.textInputLayoutCardHolder.setVisible(component.isHolderNameRequire()))`

If you like the idea I can open a new PR and do a refactor for the whole project in it.

Reply:
What about this?
```kotlin
var View.visible: Boolean
    get() = this.visibility == VISIBLE
    set(value) {
        if (value) this.visibility = VISIBLE
        else this.visibility = GONE
    }
```

Reply:
Even better! Btw you can remove `this` (I think)

Reply:
Yeah, OK, I'll add this, I was thinking if we wanted to make the view `INVISIVLE` instead, but those are less likely and we can always just set it manually.